relay-metrics/src/aggregation.rs

```diff
@@ -811,15 +820,22 @@ impl AggregatorConfig {
     ///
     /// Recent buckets are flushed after a grace period of `initial_delay`. Backdated buckets, that
     /// is, buckets that lie in the past, are flushed after the shorter `debounce_delay`.
-    fn get_flush_time(&self, bucket_timestamp: UnixTimestamp) -> Instant {
+    fn get_flush_time(&self, bucket_timestamp: UnixTimestamp, project_key: ProjectKey) -> Instant {
         let now = Instant::now();
 
         if let MonotonicResult::Instant(instant) = bucket_timestamp.to_instant() {
             let bucket_end = instant + self.bucket_interval();
             let initial_flush = bucket_end + self.initial_delay();
+            // If the initial flush is still pending, use that.
             if initial_flush > now {
-                // If the initial flush is still pending, use that.
-                return initial_flush;
+                // Shift deterministically within one bucket interval based on the project key. This
+                // distributes buckets over time while also flushing all buckets of the same project
+                // key together.
+                let mut hasher = FnvHasher::default();
+                hasher.write(project_key.as_str().as_bytes());
```

Comment:
nit: Should we add an `as_bytes()` method to `ProjectKey`?

Reply:
I was thinking about this, but the ProjectKey in fact stores a `String` (just internally as bytes), so I rather would not pollute the public API of `ProjectKey` for some internals of the metrics aggregator.